tmail-backend/jmap/extensions/src/test/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepositoryTest.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Test
+    void getAttendanceStatusShouldFallbackToNeedsActionWhenNoEventAttendanceFlagIsLinkedToMail() {
+        Flags flags = new Flags(Flags.Flag.RECENT);
+        MessageId messageId = createMessage(flags);
+
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.NeedsAction);
+    }
+
+    @Test
+    void setAttendanceStatusShouldSetAcceptedFlag() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+
+        assertThat(updatedFlags.contains("$accepted")).isTrue();
+    }
+
+    @Test
+    void setAttendanceStatusShouldSetDeclinedFlag() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Declined)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+        assertThat(updatedFlags.contains("$rejected")).isTrue();
+    }
+
+    @Test
+    void setAttendanceStatusShouldSetTentativeFlag() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Tentative)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+        assertThat(updatedFlags.contains("$tentativelyaccepted")).isTrue();
+    }
+
+    @Test
+    void setAttendanceStatusShouldRemoveExistingEventAttendanceFlags() {
+        Flags flags = new Flags("$accepted");
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Declined)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+        assertThat(updatedFlags.contains("$accepted")).isFalse();
+        assertThat(updatedFlags.contains("$rejected")).isTrue();
+    }
+
+    @Test
+    void setAttendanceStatusShouldBeIdempotent() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+
+        assertThat(updatedFlags.contains("$accepted")).isTrue();
+        assertThat(updatedFlags.getUserFlags().length).isEqualTo(1);
+    }
+
+    private Flags getFlags(MessageId messageId) {
+        return Mono.from(messageIdManagerTestSystem.getMessageIdManager()
+                .getMessagesReactive(List.of(messageId), FetchGroup.MINIMAL, session))
+            .map(MessageResult::getFlags)
+            .block();
+    }
```

Comment:
We can add one more test for:
- Given: message A with Accepted
- When: update message A with Rejected
- Then: attendee flags of message A contain only Rejected

Reply:
Such test already exist: setAttendanceStatusShouldRemoveExistingEventAttendanceFlags